### PR TITLE
Move tests to GitHub Actions

### PR DESCRIPTION
**Pull request by franmomu**
This is based on https://github.com/doctrine/mongodb-odm/pull/2250

I've changed to check just Symfony maintained versions 4.4 and 5.1.

Some doubts:

I don't know if `composer global require --no-progress --no-scripts --no-plugins symfony/flex dev-master` is necessary.
And also about this:
```yml
    # Test dev versions
    - php: 7.3
      if: type = cron
      env: DEV COMPOSER_FLAGS="-n --prefer-dist"
```

The other cases I think they are covered.

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        if: "${{ matrix.deps == 'low' }}"
+
+      - name: "Run PHPUnit"
+        run: "vendor/bin/phpunit --coverage-clover=coverage.xml"
+
+      - name: "Upload coverage file"
+        uses: "actions/upload-artifact@v2"
+        with:
+          name: "${{ matrix.deps }}-${{ matrix.php-version }}.xml"
+          path: coverage.xml
```
Comment (greg0ire) [inline .github/workflows/continuous-integration.yml]: I think you forgot another job: the one that downloads all the coverage artifacts and uploads them to Codecov.

Comment (franmomu) [reply .github/workflows/continuous-integration.yml]: Thanks! I used https://github.com/doctrine/collections/pull/258 job.

**Comment by greg0ire** [comment]
Please also update the README badge and `.gitattributes`

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
The line with `symfony/flex` was introduced by @alcaeus and @chalasr in 8fcefac455608e0f7dfc98ad6eb5de0f19990798 as part of https://github.com/doctrine/DoctrineMongoDBBundle/pull/602 . So was the line with `cron`.

**Comment by greg0ire** [comment]
@chalasr can you please review this?

**Comment by chalasr** [comment]
Installing flex globally used to speed up composer package downloads thanks to parallelized prefetching. That's not useful anymore on Composer 2

**Comment by greg0ire** [comment]
Discussed with @alcaeus about the `cron`, and we can ignore it for now.

**Comment by greg0ire** [comment]
Thanks @franmomu !

**Pull request merged**